pkg/operation/botanist/component/coredns/coredns.go

```diff
@@ -528,6 +536,138 @@ import custom/*.server
+				},
+				{
+					APIGroups: []string{""},
+					Resources: []string{"replicationcontrollers/scale"},
+					Verbs:     []string{"get", "update"},
+				},
+				{
+					APIGroups: []string{"apps"},
+					Resources: []string{"deployments/scale", "replicasets/scale"},
+					Verbs:     []string{"get", "update"},
+				},
+				// Remove the configmaps rule once below issue is fixed:
+				// kubernetes-incubator/cluster-proportional-autoscaler#16
+				{
+					APIGroups: []string{""},
+					Resources: []string{"configmaps"},
+					Verbs:     []string{"get", "create"},
+				},
+			},
+		}
+
+		clusterProportionalDNSAutoscalerClusterRoleBinding = &rbacv1.ClusterRoleBinding{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "system:coredns-autoscaler",
+			},
+			RoleRef: rbacv1.RoleRef{
+				APIGroup: rbacv1.GroupName,
+				Kind:     "ClusterRole",
+				Name:     clusterProportionalDNSAutoscalerClusterRole.Name,
+			},
+			Subjects: []rbacv1.Subject{{
+				Kind:      rbacv1.ServiceAccountKind,
+				Name:      clusterProportionalDNSAutoscalerServiceAccount.Name,
+				Namespace: clusterProportionalDNSAutoscalerServiceAccount.Namespace,
+			}},
+		}
+
+		clusterProportionalDNSAutoscalerDeployment = &appsv1.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "coredns-autoscaler",
+				Namespace: metav1.NamespaceSystem,
+				Labels:    getClusterProportionalDNSAutoscalerLabels(),
+			},
+			Spec: appsv1.DeploymentSpec{
+				Selector: &metav1.LabelSelector{
+					MatchLabels: map[string]string{LabelKey: clusterProportionalDNSAutoscalerLabelValue},
+				},
+				Template: corev1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: getClusterProportionalDNSAutoscalerLabels(),
+					},
+					Spec: corev1.PodSpec{
+						PriorityClassName:  "system-cluster-critical",
+						ServiceAccountName: clusterProportionalDNSAutoscalerServiceAccount.Name,
+						SecurityContext: &corev1.PodSecurityContext{
+							SeccompProfile: &corev1.SeccompProfile{
+								Type: corev1.SeccompProfileTypeRuntimeDefault,
+							},
+							RunAsNonRoot:       pointer.Bool(true),
+							RunAsUser:          pointer.Int64(65534),
+							SupplementalGroups: []int64{65534},
+							FSGroup:            pointer.Int64(65534),
+						},
+						Tolerations: []corev1.Toleration{{
+							Key:      "CriticalAddonsOnly",
+							Operator: corev1.TolerationOpExists,
+						}},
+						Containers: []corev1.Container{{
+							Name:            "autoscaler",
+							Image:           c.values.ClusterProportionalAutoscalerImage,
+							ImagePullPolicy: corev1.PullIfNotPresent,
+							Command: []string{
+								"/cluster-proportional-autoscaler",
+								"--namespace=" + metav1.NamespaceSystem,
+								"--configmap=coredns-autoscaler",
+								"--target=deployment/" + deployment.Name,
+								"--default-params={\"linear\":{\"coresPerReplica\":256,\"nodesPerReplica\":16,\"min\":2,\"preventSinglePointFailure\":true,\"includeUnschedulableNodes\":true}}",
```

Comment:
Nit: Use \` to avoid escaping `"`

Reply:
Changed to using ` for the line including quotes with latest commit.